lib/python/ensembl/brc4/runnable/process_gff3.py

```diff
@@ -264,15 +270,53 @@ def simpler_gff3(self, gff3_in: TextIO, out_gff_path: Path, out_funcann_path: Pa
                 new_records.append(new_record)
             
             if fail_types and not skip_unrecognized:
-                raise Exception("Unrecognized types found (%s): fail" %
-                                (" ".join(fail_types.keys())))
+                raise Exception(f"Unrecognized types found ({' '.join(fail_types.keys())}): fail")
             
             GFF.write(new_records, gff3_out)
         
         # Write functional annotation
         functional_annotation = self.clean_functional_annotations(functional_annotation)
         print_json(out_funcann_path, functional_annotation)
 
+    def format_mobile_element(self, feat, functional_annotation):
+        """Given a mobile_genetic_element feature, transform it into a transposable_element"""
+        quals = feat.qualifiers
+
+        if feat.type == 'mobile_genetic_element':
+            mobile_element_type = feat.qualifiers.get("mobile_element_type", [])
+            if mobile_element_type:
+                # Get the type (and name) from the attrib
+                if ":" in mobile_element_type:
+                    element_type, element_name = mobile_element_type[0].split(':')
+                    description = f"{element_type} ({element_name})"
+                else:
+                    element_type = mobile_element_type[0]
+                    description = element_type
+
+                # Keep the metdata in the description if the type is known
+                if element_type in ("transposon", "retrotransposon"):
+                    feat.type = 'transposable_element'
+                    if not feat.qualifiers.get("product"):
+                        feat.qualifiers["product"] = [description]
+                else:    
+                    print(f"Mobile genetic element 'mobile_element_type' is not transposon: {element_type}")
+            else:
+                print("Mobile genetic element does not have a 'mobile_element_type' tag")
+        
+        # At this point, if it is valid it should be a transposable_element feature
+        if feat.type != "transposable_element":
```

Comment:
This if assumed that some mobile_genetic_elements would not be changed to transposable_element (because the format would not be as expected), and so would return here, before adding things to the functional annotation.

I've changed the code above to return the feature directly, so that elif is now appropriate.